pkg/controller/infinispan/infinispan_controller.go

```diff
@@ -855,11 +848,17 @@ func createCacheServiceDefaultCache(podName string, infinispan *infinispanv1.Inf
             </memory>
             <partition-handling when-split="ALLOW_READ_WRITES" merge-policy="REMOVE_ALL" />
         </distributed-cache>
-    </cache-container></infinispan>`
+    </cache-container></infinispan>`, nil
+}
 
+func createCacheServiceDefaultCache(podName string, infinispan *infinispanv1.Infinispan, cluster ispnutil.ClusterInterface, logger logr.Logger) error {
+	defaultCacheXML, err := GetDefaultCacheTemplateXML(podName, infinispan, cluster, logger)
+	if err != nil {
+		return err
+	}
 	secretName := infinispan.GetSecretName()
-	protocol := infinispanProtocol(infinispan)
-	return cluster.CreateCache("default", defaultCacheXML, secretName, podName, namespace, protocol)
+	protocol := infinispan.GetEndpointScheme()
+	return cluster.CreateCache("default", defaultCacheXML, secretName, podName, infinispan.Namespace, string(protocol))
```

Comment:
not sure what's going on here but it leads me to another question. for Cache Service, we will create copies of the default cache only, right?

so in effect, for Cache Service all that is allowed is to pass cache_1.yaml which creates a copy of the default cache with a new name. if you attempt to add an XML definition or cache template, the operator throws an error?

Reply:
The change itself is just cosmetic.
The `default` cache is created by default when the cluster is deployed, user can create caches of any kind later.

Reply:
oh really? I thought users could only create copies of the default when using Cache service. so you could create caches with any definition? imo that seems to defeat the purpose of the Cache service.

Reply:
mmm, I see your concern. We can add a check that blocks CR cache creation with arbitrary template, but user will still be able to create them programmatically

Reply:
@oraNod I've added the check to block cache with templates on CacheService.

Reply:
@oraNod I've update the use case at the top with the limitation for the CacheService.

Reply:
@rigazilla thanks. that looks good!